File: Leetcode/Daily/19-05-2024/prob.go
func maximumValueSum(nums []int, k int, edges [][]int) int64 {
    var sum int64 = 0
    cnt := 0
    var sac int64 = 1e9
    for _,c := range nums{
        if (c^k)>c{
            sum += int64(c ^ k)
            sac = min(sac,int64((c ^ k) - c))
            cnt++
        }else{
            sum += int64(c)
            sac = min(sac,int64(c-(c^k)))
        }
    }

    if cnt%2 == 0{
        return sum
    }

    return sum - sac
}

File: Leetcode/Daily/19-05-2024/prob.cpp
class Solution {
public:
    long long maximumValueSum(vector<int>& nums, int k, vector<vector<int>>& edges) {
        long long sum = 0;
        int sac = 1e9;
        int cnt = 0;
        for(int i=0;i<nums.size();i++){
            if( (nums[i]^k) >nums[i] ){
                sum += (nums[i]^k);
                sac = min(sac, (nums[i]^k) -nums[i] );
                cnt++;
            }
            else{
                sum += nums[i];
                sac = min(sac, nums[i]-(nums[i]^k));
            }
        }

        if(cnt%2==0)return sum;

        return sum - sac;
    }
};